tests/framework/queries.go

```diff
@@ -0,0 +1,198 @@
+package framework
```

Comment:
I created this as I see a possible opportunity to use the same prometheus queries in the reconfiguration test, the current ones in the `scale_test.go`, and any future tests. 

However, the ones in the `scale_test.go` use queries that have a start time in the query. Something like:

```
eventsAvgTime := getFirstValueOfVector(
			fmt.Sprintf(
				`(nginx_gateway_fabric_event_batch_processing_milliseconds_sum{pod="%[1]s"}`+
					` - `+
					`nginx_gateway_fabric_event_batch_processing_milliseconds_sum{pod="%[1]s"} @ %[2]d)`+
					` / `+
					`(nginx_gateway_fabric_event_batch_processing_milliseconds_count{pod="%[1]s"}`+
					` - `+
					`nginx_gateway_fabric_event_batch_processing_milliseconds_count{pod="%[1]s"} @ %[2]d)`,
				ngfPodName,
				startTime.Unix(),
			),
		)
```

However, I ran into a few issues doing that with the reconfiguration test where the test when the resources were deployed first then NGF was deployed would not work with this type of query. So instead of doing this query with a start time in 2/3 of the tests, then these queries without the start time in the other 1/3 of the tests, I just decided to remove the start time part of the query which should be fine as we restart NGF between each test. 

Michael had added the start time part of the query as he didn't want to capture some of the data from NGF starting up, but I think that we actually want that data. 

The issue is figuring out how to re-use the code. 

1. I could do stuff like  GetReloadAvgTimeWithStartTime and  GetReloadAvgTime
2. Or just not re-use the code and keep the queries separated. 


Does anyone have any thoughts on this?

Reply:
Seems fine to me

Reply:
>Having a fresh install of NGF between tests and removing the start time from the query seems OK to me.

This would just be for this reconfiguration test right? Would it be alright for me to create queries with a start time (for the scale_test.go so i don't modify that), and queries without a start time (for this one).

Or by what you stated, are you recommending me to modify the scale test and remove the start time from the queries.

Reply:
> Would it be alright for me to create queries with a start time (for the scale_test.go so i don't modify that), and queries without a start time (for this one).

Sure, that's fine. 

> Or by what you stated, are you recommending me to modify the scale test and remove the start time from the queries.

Let's not modify the scale tests. We can file a ticket to look into refactoring the prom queries if you think we can make changes so that they can be reused across both tests.

Reply:
Yeah, we should do a full run of the NFR tests before merging this